packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -81,9 +88,22 @@ export class SummarizeHeuristicRunner implements ISummarizeHeuristicRunner {
         }
     }
 
-    public shouldRunLastSummary(): boolean {
+    public shouldRunLastSummary(stopReason: SummarizerStopReason): boolean {
+        if (stopReason !== "parentNotConnected") {
+            return false;
+        }
+
         const opsSinceLastAck = this.opsSinceLastAck;
-        return (opsSinceLastAck > this.minOpsForAttemptOnClose);
+        const minOpsForAttemptOnClose = this.minOpsForAttemptOnClose;
+
+        this.logger.sendTelemetryEvent({
+            eventName: "ShouldRunLastSummary",
+            stopReason,
+            opsSinceLastAck,
+            minOpsForAttemptOnClose,
+        });
+
+        return opsSinceLastAck > minOpsForAttemptOnClose;
```

Comment:
This should probably be >=. Not a big deal but the name "minOpsForAttemptOnClose" would suggest that we would summarize it we hit this number.